commit 0b40f996a572fb1f00b7dcf84eb226b78e9122ce
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-06-18T19:00:38+00:00 (1 year, 4 months ago)
    
    joyent/node-verror#56 Add more tests to tst.context.js

diff --git a/test/tst.context.js b/test/tst.context.js
index 300a4a3..7623a73 100644
--- a/test/tst.context.js
+++ b/test/tst.context.js
@@ -4,21 +4,35 @@
 
 var mod_assert = require('assert');
 var mod_verror = require('../lib/verror');
-var mod_isError = require('core-util-is').isError;
 var mod_vm = require('vm');
 
 var VError = mod_verror.VError;
 var WError = mod_verror.WError;
 
-var err = new Error();
-var verr = new VError(err);
-mod_assert.ok(mod_isError(verr.cause()));
+var prog1 = 'callback(new Error(), "Error")';
+var prog2 = 'var e = new Error(); e.name = "BarError"; callback(e, "BarError")';
+
+function runTests(cerr, name) {
+	var verr;
+
+	verr = new VError(cerr);
+	mod_assert.equal(verr.cause(), cerr);
+
+	verr = new VError({ cause: cerr });
+	mod_assert.equal(verr.cause(), cerr);
+
+	mod_assert.deepEqual(mod_verror.cause(cerr), null);
+	mod_assert.deepEqual(mod_verror.info(cerr), {});
+
+	mod_assert.deepEqual(mod_verror.findCauseByName(cerr, 'FooError'), null);
+	mod_assert.equal(mod_verror.findCauseByName(verr, name), cerr);
+
+	mod_assert.equal(typeof (mod_verror.fullStack(cerr)), 'string');
+}
 
 var context = mod_vm.createContext({
-	'callback': function callback(err2) {
-		mod_assert.ok(mod_isError(err2));
-		var verr2 = new VError(err);
-		mod_assert.ok(mod_isError(verr2.cause()));
-	}
+	'callback': runTests
 });
-mod_vm.runInContext('callback(new Error())', context);
+
+mod_vm.runInContext(prog1, context);
+mod_vm.runInContext(prog2, context);
